atlp-rwanda/atlp-devpulse-fn
# A user will be able to react and comment to a blog post

**PR Description**

**Feature:** A user should be able to comment and react to blog posts.

**What does this PR do?**

This PR implements frontend functionality for:

- Adding comments to a blog post.
- Reacting to blog posts with reactions (e.g., Like, Love, Support, Funny, Celebrate).
- Liking individual comments.
- Replying to the blog comments.

**Related Issues**
N/A

**Changes Made:**

- Added reactionActions to handle reaction-related functionalities
- Added commentActions to handle comment-related functionalities.
- Created reactionReducers to manage the state for reactions.
- Created commentReducers to manage the state for comments.
- Developed the BlogComment component to display and manage blog post comments.
- Developed the CommentLike component to allow liking comments.
- Developed the BlogReactions component to display and manage reactions on blog posts.
- Modified the SingleBlog component to integrate the new components (BlogComment, CommentLike, and BlogReactions).

**How to Test This PR:**

- Clone the repository.
- Run npm install to install dependencies and navigate to the appropriate the branch ft-comments-reactions-fn.
- Start the server using npm run dev.
- Confirm that the single blog view displays comments and reactions correctly, and that interactions are functional and reflected in the UI.

**Screenshoots**

<img width="765" alt="image" src="https://github.com/user-attachments/assets/9ef4b50d-6951-4f40-b52c-65db908eafe0">
<img width="231" alt="{CD6BFC1A-706B-4972-8FAA-FBB4162CAE80}" src="https://github.com/user-attachments/assets/76bc4421-cf2a-4883-8593-acc54ef7e487">
<img width="438" alt="{5691E666-1893-4CFF-8BE7-0B9375FC26A7}" src="https://github.com/user-attachments/assets/5ccf537d-489c-43ba-a2f5-65e0040e8f27">
